ban-cooking.lua

```diff
@@ -0,0 +1,440 @@
+-- convenient way to ban cooking categories of food
+-- based on ban-cooking.rb by Putnam: https://github.com/DFHack/scripts/pull/427/files
+-- Putnams work completed by TBSTeun
+--[====[
+
+ban-cooking
+===========
+A more convenient way to ban cooking various categories of foods than the
+kitchen interface.  Usage:  ``ban-cooking <type>``.  Valid types are ``booze``,
+``honey``, ``tallow``, ``oil``, ``seeds`` (non-tree plants with seeds),
+``brew``, ``fruit``, ``mill``, ``thread``, and ``milk``.
+
+]====]
+
+kitchen = df.global.ui.kitchen
+
+already_banned = {}
+
+local function already_banned_has_key(key)
+    for k, v in pairs(already_banned) do
+        if k[1] == key[1] and k[2] == key[2] and k[3] == key[3] and k[4] == key[4] then
+            return true
+        end
+    end
+
+    return false
+end
+
+-- Iterate over the elements of the kitchen.item_types list
+for i = 0, #kitchen.item_types - 1 do
+    -- Check if the kitchen.exc_types[i] element is equal to :Cook
+    if kitchen.exc_types[i] == df.kitchen_exc_type.Cook then
+        -- Add a new element to the already_banned dictionary
+        already_banned_key = {kitchen.mat_types[i], kitchen.mat_indices[i], kitchen.item_types[i], kitchen.item_subtypes[i]}
+        if not(already_banned_has_key(already_banned_key)) then
+            already_banned[already_banned_key] = true
+        end
+    end
+end
+
+local function reaction_product_id_contains(reaction_product, str)
+    for _, s in ipairs(reaction_product.id) do
+        if s.value == str then
+            return true
+        end
+    end
+
+    return false
+end
+
+local function ban_cooking(print_name, mat_type, mat_index, type, subtype)
+    key = mat_type, mat_index, type, subtype
+    -- Skip adding a new entry further below, if the item is already banned.
+    if already_banned[key] then
+        return
+    end
+    -- The item hasn't already been banned, so we do that here by appending its values to the various arrays
+    print(print_name .. ' has been banned!')
+    kitchen.mat_types:insert('#', mat_type)
+    kitchen.mat_indices:insert('#', mat_index)
+    kitchen.item_types:insert('#', type)
+    kitchen.item_subtypes:insert('#', subtype)
+    kitchen.exc_types:insert('#', df.kitchen_exc_type.Cook)
+    already_banned[key] = true
+end
+
+local funcs = {}
+
+funcs["booze"] = function()
```

Comment:
`funcs["booze"]` can be simplified to just `funcs.booze`. you only need to use the first form if the string is not a valid identifier (like "hyphen-word")